microcluster/app.go

```diff
@@ -255,12 +255,45 @@ func (m *MicroCluster) RecoverFromQuorumLoss(members []cluster.LocalMember) erro
 		nodeInfo = append(nodeInfo, *info)
 	}
 
-	// Check each cluster member's /1.0 endpoint to ensure that they are unreachable
+	// Check each cluster member's /1.0 to ensure that they are unreachable.
 	// This is a sanity check to ensure that we're not reconfiguring a cluster
 	// that's still partially up.
-	// It may also be possible to check the raft term of each surviving member
-	// and redirect the user to call RecoverFromQuorumLoss on that member instead.
-	// TODO
+	remotes, err := recover.ReadTrustStore(m.FileSystem.TrustDir)
+	if err != nil {
+		return err
+	}
+
+	serverCert, err := m.FileSystem.ServerCert()
+	if err != nil {
+		return err
+	}
+
+	clusterCert, err := m.FileSystem.ClusterCert()
+	if err != nil {
+		return err
+	}
+
+	clusterKey, err := clusterCert.PublicKeyX509()
+	if err != nil {
+		return err
+	}
+
+	cluster, err := remotes.Cluster(false, serverCert, clusterKey)
+	if err != nil {
+		return err
+	}
+
+	err = cluster.Query(context.Background(), true, func(ctx context.Context, client *client.Client) error {
+		var rslt internalTypes.Server
+		err := client.Query(ctx, "GET", "1.0", api.NewURL(), nil, &rslt)
```

Comment:
What is the default context timeout for this query? Should we specify a rather short one?

Reply:
I reduced it to 10s so the user isn't stuck waiting for half a minute if for some reason we don't get a connection refused but the server we're connected to doesn't come back with a response. It doesn't look like `GET /1.0` queries the DB, so I'm a little unsure why we'd be able to establish a connection but not get a response (unless a machine has too much memory or CPU pressure).
This bug is old so it might not be relevant anymore, but I'm also not convinced that we can rely on "Context deadline exceeded" meaning that we've successfully established a connection: https://github.com/golang/go/issues/31863
Happy to make more changes here if needed.